src/proposals/20250529/SparkEthereum_20250529.t.sol

```diff
@@ -0,0 +1,796 @@
+
+        vm.stopPrank();
+
+        chainData[ChainIdUtils.Optimism()].domain.selectFork();
+
+        // Optimism Sky Core spell configuration
+        vm.startPrank(Optimism.SKY_GOV_RELAY);
+
+        IOptimismTokenBridge(Optimism.TOKEN_BRIDGE).registerToken(Ethereum.USDS, Optimism.USDS);
+        IOptimismTokenBridge(Optimism.TOKEN_BRIDGE).registerToken(Ethereum.SUSDS, Optimism.SUSDS);
+        IAuthLike(Optimism.USDS).rely(Optimism.TOKEN_BRIDGE);
+        IAuthLike(Optimism.SUSDS).rely(Optimism.TOKEN_BRIDGE);
+
+        vm.stopPrank();
+
+        chainData[ChainIdUtils.Unichain()].domain.selectFork();
+
+        // Unichain Sky Core spell configuration
+        vm.startPrank(Unichain.SKY_GOV_RELAY);
+
+        IOptimismTokenBridge(Unichain.TOKEN_BRIDGE).registerToken(Ethereum.USDS, Unichain.USDS);
+        IOptimismTokenBridge(Unichain.TOKEN_BRIDGE).registerToken(Ethereum.SUSDS, Unichain.SUSDS);
+        IAuthLike(Unichain.USDS).rely(Unichain.TOKEN_BRIDGE);
+        IAuthLike(Unichain.SUSDS).rely(Unichain.TOKEN_BRIDGE);
+
+        vm.stopPrank();
+
+        chainData[ChainIdUtils.Ethereum()].domain.selectFork();
+    }
+
+    function test_ETHEREUM_optimismCctpConfiguration() public onChain(ChainIdUtils.Ethereum()) {
+        bytes32 optimismKey = RateLimitHelpers.makeDomainKey(
+            MainnetController(Ethereum.ALM_CONTROLLER).LIMIT_USDC_TO_DOMAIN(),
+            CCTPForwarder.DOMAIN_ID_CIRCLE_OPTIMISM
+        );
+
+        _assertRateLimit(optimismKey, 0, 0);
+        assertEq(MainnetController(Ethereum.ALM_CONTROLLER).mintRecipients(CCTPForwarder.DOMAIN_ID_CIRCLE_OPTIMISM), bytes32(0));
+
+        executeAllPayloadsAndBridges();
+
+        _assertRateLimit(optimismKey, 50_000_000e6, 25_000_000e6 / uint256(1 days));
+        assertEq(MainnetController(Ethereum.ALM_CONTROLLER).mintRecipients(CCTPForwarder.DOMAIN_ID_CIRCLE_OPTIMISM), bytes32(uint256(uint160(Optimism.ALM_PROXY))));
+    }
+
+    function test_OPTIMISM_almControllerDeployment() public onChain(ChainIdUtils.Optimism()) {
+        // Copied from the init library, but no harm checking this here
+        IALMProxy         almProxy   = IALMProxy(Optimism.ALM_PROXY);
+        IRateLimits       rateLimits = IRateLimits(Optimism.ALM_RATE_LIMITS);
+        ForeignController controller = ForeignController(Optimism.ALM_CONTROLLER);
+
+        assertEq(almProxy.hasRole(0x0,   Optimism.SPARK_EXECUTOR), true, "incorrect-admin-almProxy");
+        assertEq(rateLimits.hasRole(0x0, Optimism.SPARK_EXECUTOR), true, "incorrect-admin-rateLimits");
+        assertEq(controller.hasRole(0x0, Optimism.SPARK_EXECUTOR), true, "incorrect-admin-controller");
+
+        assertEq(almProxy.hasRole(0x0,   DEPLOYER), false, "incorrect-admin-almProxy");
+        assertEq(rateLimits.hasRole(0x0, DEPLOYER), false, "incorrect-admin-rateLimits");
+        assertEq(controller.hasRole(0x0, DEPLOYER), false, "incorrect-admin-controller");
```

Comment:
There are two deployer addresses. I did the Optimism ones so my address should be checked. Otherwise this is a useless check.

Reply:
added your deployer address too